### DISCUSSION TITLE
Conference Solution Improvemets

### DISCUSSION THREAD

**Discussion by burak-58** [ant-media/Ant-Media-Server#5960] [General]
Hello Dear Ant Media Server users, especially conference solution enthusiasts,

Exciting news awaits you!

Since the inception of our conference solution, we've been dedicated to enhancing its capabilities. We began with the introduction of the stream-based conference solution and later unveiled the track-based conference solution, also known as multitrack-conference. If you're curious to learn more, check out our [blog post](https://antmedia.io/reveal-the-secrets-of-3-types-of-video-conference-solutions).

Now, we're thrilled to introduce an improved version of the multitrack conference, set to be available after version 2.8.0 (already accessible in the latest snapshots). Let's explore the enhancements together:

1. **New Websocket Message:** We've introduced a new websocket message called `getBroadcastObject` to retrieve the Broadcast structure as JSON. This broadcast can be either a Maintrack or a Subtrack. Obtain a Broadcast object for a main track by calling `getBroadcastObject` with the room ID, and for a subtrack, use its ID.

2. **Room Data Structure Update:** We no longer utilize the Room data structure. Instead, we keep participant IDs as subtracks of maintracks. There's no need to call `JoinTheRoom` or `LeaveTheRoom` methods anymore.

3. **Event-Based Approach:** Say goodbye to polling the `getRoomInfo` method. We've introduced an event-based approach. Applications will now be notified by the `TRACK_LIST_UPDATED` Data Channel message. Upon receiving this notification, applications can call `getBroadcastObject` for the Maintrack, check the subtracks, and obtain participant names by calling `getBroadcastObject` for the participant ID.

4. **Metadata Field Usage:** As a best practice, you can use the Metadata field in the Broadcast object to persist extra information (e.g., camera/mic status, screen sharing status) for a participant.

5. **Updated Sample Conference Pages:** With these improvements, we've also updated the sample conference pages in the deployment:
   - `conference.html` is the sample implementation for this new structure.
   - `maintrack-conference.html` redirects to `conference.html`.
   - `conference-deprecated.html` serves as a sample for the oldest (stream-based) solution. (Previously, `conference.html` was used as a sample for this solution.)

Feel free to explore and implement these enhancements in your conference solutions. We're excited about the possibilities these improvements bring to your experience with Ant Media Server's conference solution.

**Comment by FarhodKurbonov** [comment]
Hi @burak-58 , we have started implementing reconnection across all our platforms (Android, iOS, Web) following your and @mustafaboleken suggestion from [here](https://github.com/orgs/ant-media/discussions/6254). The way how of `TRACK_LIST_UPDATED` is being triggered not meeting user expectation. We tested your implementation using the official [GitHub repository for the Ant Media Video Conferencing application](https://github.com/ant-media/conference-call-application/tree/main), both locally and on your deployed [Circle App](https://meet.antmedia.io/Conference/).

We reviewed the AntMedia.js file in the Circle App and noticed that when a participant's internet connection is dropped, the server waits for him (does not close the websocket connection and session). However, `TRACK_LIST_UPDATED` **is not triggered on the side of other participants**. It only triggers when a client app (the one who has been having trouble with reconnection) drops the connection (closes a web browser tab or kills the app), which is predictable. In this scenario, the server closes the websocket connection and  session with the room participant and triggers `TRACK_LIST_UPDATED`.

Is there any way for server to notify room participants about  those ones who lost connection and tries to get connected?

 Steps to reproduce:

* `User A` joins the room (abc).
* `User B` joins the same room.
* `User A` turns off or physically unplugs his internet connection (WiFi/4G, etc.).

Actual Behavior:
When the internet is disconnected on `User A's` side, the app tries to reconnect and notifies `User A` which is perfect. Meanwhile, `User B` is unaware of what is happening on `User A's` side, and it appears as if `User A` is still online

Expected Behaviour:
When the internet is disconnected on `User A's` side, the app should notify `User B` about it. This is important because `User B` may continue speaking to `User A`, not realising that he is offline and trying to reconnect. The app should also display the reconnection status of `User A` ON `User B's` side.

Additional Info:

Server Version 2.8.2 Enterprise 
Tested version of Circle App: 2.8.2 and the version officially deployed on meet.antmedia.io.

**Reply by yashtandon113** [reply]
@FarhodKurbonov 
As I see, you have created another discussion thread, so the team will reply to you on the same.
https://github.com/orgs/ant-media/discussions/6330

**Reply by FarhodKurbonov** [reply]
Thank you @yashtandon113
